e2e-tests/functions

```diff
@@ -564,7 +564,7 @@ compare_mysql_cmd() {
 
 	run_mysql "$command" "$uri" \
 		>$tmp_dir/${command_id}.sql
-	if [ ! -s "$tmp_dir/${command_id}.sql" ]; then
+	if [[ ! -s "$tmp_dir/${command_id}.sql" || $(cat "$tmp_dir/${command_id}.sql" | grep 'Unknown MySQL server host') ]]; then
```

Comment:
grep supports checking files naitively, we can skip cat

![Screenshot 2025-04-11 at 1 50 52 PM](https://github.com/user-attachments/assets/a7641d20-31cc-4810-ba85-a85161252c94)